compat/maven-settings-builder/src/main/java/org/apache/maven/settings/building/DefaultSettingsBuilder.java

```diff
@@ -251,4 +255,29 @@ public Object execute(String expression, Object value) {
 
         return result;
     }
+
+    private List<Server> serversByIds(List<Server> servers) {
+
+        if (servers.stream().allMatch(server -> server.getIds().isEmpty())) {
+            return servers;
+        }
+
+        return servers.stream()
+                .flatMap(server -> {
+                    List<String> ids = server.getIds();
+                    if (ids.isEmpty()) {
+                        return Stream.of(server);
+                    } else {
+                        return Stream.concat(Stream.of(server), ids.stream().map(id -> newServer(server, id)));
+                    }
+                })
+                .collect(Collectors.toList());
```

Comment:
is this API possible?
```suggestion
                .toList();
```
- https://docs.openrewrite.org/recipes/java/migrate/util/replacestreamcollectwithtolist
- https://github.com/apache/maven/pull/2322

Reply:
will give no more examples. Its clear we produce outdated code. Need proper tooling.